features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/DocumentEnricher.java

```diff
@@ -150,40 +148,24 @@ private static boolean isPrivateAddress(String ipAddress) {
         return inetAddress.isLoopbackAddress() || inetAddress.isLinkLocalAddress() || inetAddress.isSiteLocalAddress();
     }
 
-    private Optional<NodeDocument> getNodeInfoFromCache(final String location, final String ipAddress, final ContextKey contextKey, final String value) {
-        final NodeInfoKey key = new NodeInfoKey(location, ipAddress, contextKey, value);
-        try {
-            return nodeInfoCache.get(key);
-        } catch (ExecutionException e) {
-            LOG.error("Error while retrieving NodeDocument from NodeInfoCache: {}.", e.getMessage(), e);
-            throw new RuntimeException(e);
+    private Optional<NodeDocument> getNodeInfoFromCache(final String location, final String ipAddress) {
+        final Optional<Integer> nodeId = interfaceToNodeCache.getFirstNodeId(location, InetAddressUtils.addr(ipAddress));
+        if(nodeId.isPresent()) {
+            try {
+                return nodeInfoCache.get(nodeId.get());
+            } catch (ExecutionException e) {
+                LOG.error("Error while retrieving NodeDocument from NodeInfoCache: {}.", e.getMessage(), e);
+                throw new RuntimeException(e);
+            }
         }
+        return Optional.empty();
     }
 
-    private Optional<NodeDocument> getNodeInfo(final String location, final String ipAddress, final ContextKey contextKey, final String value) {
-        return getNodeInfo(location, InetAddressUtils.addr(ipAddress), contextKey, value);
-    }
 
-    private Optional<NodeDocument> getNodeInfo(final String location, final InetAddress ipAddress, final ContextKey contextKey, final String value) {
+    private Optional<NodeDocument> getNodeInfo(Integer nodeId) {
         OnmsNode onmsNode = null;
-
-        if (contextKey != null && !Strings.isNullOrEmpty(value)) {
```

Comment:
We need to keep this logic. We can perform the check before the `interfaceToNodeCache` lookup, and store the results in a separate cache (keyed by `contextKey`).